### PR TITLE
docker run parameters

### PR DESCRIPTION
**Pull request by rpitonak**
based on work done in this [PR](https://github.com/dhodovsk/conu/pull/2)

Now we can convert object of type `DockerRunBuilder` and create instance of new introduced class `DockerContainerParameters`. Properties of this class can be used for running `docker run` using [docker-py API](https://github.com/docker/docker-py).

Functionality not cover all possible parameters, but they can be implemented easily in future PRs if this changes will be merged.

I will be glad for review.

### CODE REVIEW

**Comment by rpitonak** [comment]
Thank you for the review. I pushed new changes.

**Comment by TomasTomecek** [comment]
This is also worth discussing. I would personally prefer only the minimum amount of parameters to be part of the metadata definition and the rest be just an extension. We don't need to 1:1 copy docker, we can just get inspired.

File: conu/backend/docker/container_parameters.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+class DockerContainerParameters(object):
```
Comment (dhodovsk) [inline conu/backend/docker/container_parameters.py]: Use ContainerMetadata as parent.

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/F0VfV](https://da.gd/F0VfV)__
* Comment: __See details for logs__

File: conu/backend/docker/image.py [hunk 2]
```diff
@@ -393,6 +393,47 @@ def callback():
             container_name = actual_name
         return DockerContainer(self, container_id, popen_instance=popen_instance, name=container_name)
 
+    def run_via_api(self, container_params):
+        """
+        :param container_params:
```
Comment (TomasTomecek) [inline conu/backend/docker/image.py]: please document

File: conu/backend/docker/image.py [hunk 3]
```diff
@@ -393,6 +393,47 @@ def callback():
...
+        :return: instance of DockerContainer
+        """
+
+        client = docker.from_env()
+        container = client.containers.run(self.get_id(), command=container_params.command,
```
Comment (TomasTomecek) [inline conu/backend/docker/image.py]: please use the low level API which is available at `self.d`

**Comment by TomasTomecek** [comment]
Github tells me now that both of you authored the commit. I would say that's sufficient.

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/bgvvu](https://da.gd/bgvvu)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/CKsY](https://da.gd/CKsY)__
* Comment: __See details for logs__

**Comment by TomasTomecek** [comment]
This looks pretty solid, ready to review&merge? @dhodovsk PTAL

File: conu/backend/docker/image.py [hunk 4]
```diff
@@ -384,6 +384,53 @@ def callback():
...
         return DockerContainer(self, container_id, popen_instance=popen_instance, name=container_name)
 
+    def run_via_api(self, container_params):
+        """
+        create a container using this image and run it in background via Docker-py API.
+        https://docker-py.readthedocs.io/en/stable/api.html
+        :param container_params: DockerContainerParameters
+        :return: instance of DockerContainer
+        """
+
```
Comment (dhodovsk) [inline conu/backend/docker/image.py]: To be able to run container with `run_via_api()` without commands or parameters, please add
```
if not container_params:
     container_params = DockerContainerParameters()
```

File: conu/backend/docker/container.py [hunk 5]
```diff
@@ -70,6 +72,161 @@ def build(self):
...
+        parser.add_argument("--no-healthcheck", action="store_true", dest="no_healthcheck")
+
+        args = parser.parse_args(args=self.options)
+        command = self.arguments
+
+        options_dict = vars(args)
+
+        # create Healthcheck object
+        if not options_dict.pop("no_healthcheck", None):
+            options_dict["healthcheck"] = Healthcheck(
```
[the remark this answers is not in the document]
Comment (TomasTomecek) [reply conu/backend/docker/container.py]: Does anyone actually use healthchecks? :D

I would probably just ignore it.

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/FUz6](https://da.gd/FUz6)__
* Comment: __See details for logs__

**Comment by dhodovsk** [comment]
LGTM, please rebase so we can merge :)

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/iARl](https://da.gd/iARl)__
* Comment: __See details for logs__

**Pull request merged**